crates/rattler_conda_types/src/match_spec/parse.rs

```diff
@@ -1375,4 +1425,39 @@ mod tests {
             .collect::<Vec<_>>();
         assert_eq!(specs, parsed_specs);
     }
+
+    #[cfg(feature = "extras")]
+    #[test]
+    fn test_simple_extras() {
+        let spec = MatchSpec::from_str("foo[extras=[bar]]", Strict).unwrap();
+
+        assert_eq!(spec.extras, Some(vec!["bar".to_string()]));
+        assert!(MatchSpec::from_str("foo[extras=[bar,baz]", Strict).is_err());
+    }
```

Comment:
Lets also add some test cases for failing cases. E.g. `foo=[extras=]` or `foo=[extras=bar,baz]` or `foo=[extras=[bar,baz]`, `foo=[extras=[bar,]]`